scripts/ci/ws12_workflow_contracts.py

```
@@ -183,6 +193,46 @@ def validate_e2e_scope_filters(text: str) -> list[str]:
+
+def validate_code_style_docs_guard_order(text: str) -> list[str]:
+    """Return every way the docs-gate guard could sit past the early exit."""
+
+    guard = text.find(CODE_STYLE_DOCS_GUARD_MARKER)
+    early_exit = text.find(CODE_STYLE_HAS_CODE_EXIT_MARKER)
+    if guard == -1 or early_exit == -1:
+        # Presence itself is REQUIRED_MARKERS' job; report only what this pin
+        # cannot delegate — a missing anchor makes the order unassertable.
+        return [
+            f"{CODE_STYLE_WORKFLOW}: docs-gate guard order unassertable — "
+            f"missing {'guard' if guard == -1 else 'has_code early-exit'} marker"
```

Ruff (0.16.1)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)